packages/entity-browser/src/components/SearchForm/SearchForm.js

```diff
@@ -13,14 +13,16 @@ const SearchForm = props => {
     props.setSearchInput()
   }
 
-  const msg = id => {
-    return props.intl.formatMessage({
-      id
-    })
+  const msg = id => (props.intl.formatMessage({id}))
+
+  const isHidden = name => {
+    const field = props.preselectedSearchFields.find(f => f.id === name)
+    return field !== undefined && field.hidden
   }
 
-  const showField = name => (
-    props.disableSimpleSearch || props.simpleSearchFields.includes(name) || props.showExtendedSearchForm
+  const shouldRenderField = name => (
+    !isHidden(name) && (props.disableSimpleSearch || props.simpleSearchFields.includes(name)
+    || props.showExtendedSearchForm)
```

Comment:
* Could be improved by swapping `simpleSearchFields.includes(name)` and `props.showExtendedSearchForm`, since checking the `showExtendedSearchForm` boolean is cheaper than the `includes()` call.
* IMO the readability of the condition would be a bit better like this:
  ```
  const shouldRenderField = name =>
    !isHidden(name) && (
      props.disableSimpleSearch || props.showExtendedSearchForm || props.simpleSearchFields.includes(name)
    )